### PR TITLE
fix(tools): validate and lint i18n schema

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Related to https://github.com/freeCodeCamp/freeCodeCamp/pull/40585#issuecomment-752731620

- Removes the schema validation from the `test` call
- Adds a custom validation script for the translation objects
  - Script verifies that each language's translation objects are not missing any keys in the schema
  - Script verifies that each language's translation objects do not have keys that are missing from the schema.
  - Script verifies that no object's values are empty or falsy
  - Script verifies that the `motivation.json` object's quotes are not malformed (should have `quote` and `author` properties).
- Adds `lint` script to the `client/package.json` to call validation script
- Adds `lint:translations` script to the *root* `package.json` to call client `lint` script (this ensures it runs in our CI tests).
- Script will throw errors to fail CI if validation does not pass - error message will contain information on why validation failed.

### CODE REVIEW

**Review by raisedadead — APPROVED** [review]

File: client/i18n/motivation-schema.js [hunk 1]
```diff
@@ -1,22 +1,10 @@
 /* eslint-disable camelcase */
-/* This is used for testing to make sure the motivation.json files
- * in each language have the correct structure
+/* This is used for testing. If a translations.json file doesn't match the
```
Comment (moT01) [inline client/i18n/motivation-schema.js:2]:
```suggestion
/* This is used for testing. If a motivation.json file doesn't match the
```

**Review by moT01 — COMMENTED** [review]
I dunno, I feel like the other method was quite a bit simpler with all the same checks but the one for an empty field, which could have probably been added quite easily.

Wondering if we can or should use the english json files as the schema - I think that could be beneficial. Instead of having to add a key to the schema, then all the translations when you want to add something - it could just be added to all the json files. And we could get rid of a few of the schema files. I might be overlooking something though.

Found one minor typo...

**Comment by nhcarrigan** [comment]
> Wondering if we can or should use the english json files as the schema 

This is definitely doable. My only thought against this would be that using the separate schema file creates a single source of truth. If we forget to remove an unused key from the English file, currently the CI will say the English file has an extra key. If we shift the script to use that English file as the base, then the Chinese and Spanish files would be *missing* a key.

**Comment by raisedadead** [comment]
Hey @moT01 is this good to merge? If yes, do mind landing this.

**Review by moT01 — APPROVED** [review]
My concern was that this seemed a little more complex than the other method @raisedadead - my opinion anyway. It's a lot more code to make the same checks (except for the one mentioned I think), it splits the testing up into two scripts/areas, and if another file is added, we need to make sure to add another function to the new `schema-validation` file. The code is very well written, however, so it's not exactly difficult to add another function - and the error messages are a lot cleaner. So I think there's some pros and cons. Just trying to clarify my concerns. That being said, I'm fine with it - so lets just go with it.

**Pull request merged**